src/LeanCode.Contracts/NotificationEnvelope.cs

```diff
@@ -0,0 +1,24 @@
+namespace LeanCode.Contracts;
+
+public sealed class NotificationEnvelope
```

Comment:
I miss ctor here. :)

Reply:
Public or restricted? With explicitly passed ID but types still interfered, I suppose? Do we want to have some type safety with reflection there?

Reply:
I've added a simple one without types check, let me know if this is what you've missed. :)

Reply:
That's exactly what I had in mind. :)